### PR TITLE
Skip pipelines without triggers from displaying

### PR DESCRIPTION
**Pull request by lbarcziova**

### CODE REVIEW

**Review by Venefilyn — COMMENTED** [review]
Feels strange that triggers can be undefined. Could you update the `apiDefinitions.ts` as well? It might catch a few more errors we should catch, like in the pipeline detail

File: frontend/src/components/pipeline/PipelinesTable.tsx [hunk 1]
```diff
@@ -118,7 +118,9 @@ export function PipelinesTable() {
 
   const mappedRows = useMemo(
     () =>
-      rows.map((run) => <PipelinesTableTr key={run.merged_run_id} run={run} />),
+      rows
+        .filter((run) => run.trigger)
+        .map((run) => <PipelinesTableTr key={run.merged_run_id} run={run} />),
```
Comment (Venefilyn) [inline frontend/src/components/pipeline/PipelinesTable.tsx:123]: Checked into things a bit more, I think we can do better by displaying the rows but simply updating the `<PipelinesTableTr>`:

```tsx
<strong>
  {run.trigger && "project_url" in run.trigger ? (
    <ForgeIcon url={run.trigger.project_url} />
  ) : (
    <ForgeIconByForge />
  )}{" "}
  <Link to={`/pipeline/${run.merged_run_id}`}>
    {run.trigger && "project_url" in run.trigger ? (
      <>
        <TriggerSuffix trigger={run.trigger} />
      </>
    ) : (
      <>{run.merged_run_id}</>
    )}
  </Link>
</strong>
```

**Review by Venefilyn — CHANGES_REQUESTED** [review]

File: frontend/src/components/pipeline/PipelinesTable.tsx (hunk 1 above)
Comment (lbarcziova) [reply frontend/src/components/pipeline/PipelinesTable.tsx:123]: that was my initial solution, but I believe these are really occasionally malformed data that should not be displayed at all

Comment (Venefilyn) [reply frontend/src/components/pipeline/PipelinesTable.tsx:123]: Ack! I think this should be fixed at the service level as well to ensure this won't happen

**Review by Venefilyn — APPROVED** [review]
LGTM

**Comment by lachmanfrantisek** [comment]
@lbarcziova ,  @Venefilyn Thanks both of you!

**Pull request merged**